### ISSUE TITLE
installL and installR?

### ISSUE THREAD

**Issue by garyb** [slamdata/purescript-halogen#158]
I think these might be more useful if the `Either` extended over the state and query algebra too, I can't really think of a case where I'd want to have two types of placeholder installing the same type of components in a parent.

The way the types are now this is kinda possible (as there's nothing to stop you using `Either sl sr` in place of `s'`, and likewise for `f'`), but it needs some additional helper functions so that the child component types don't have to be defined with Either in these places.

Ideally `installL` and `installR` would let you combine something like this:

``` purescript
child1 :: Component State1 Input1 g p
child2 :: Component State2 Input2 g p
parent :: ParentComponent StateP (Either State1 State2) InputP (Either Input1 Input2) g (Const Void) (Either Placeholder1 Placeholder2) p
```

I propose making the `Either`ness internal to the installers as we can safely make the `Left`/`Right` bijections required to combine things this way - they will rely on the partial `fromLeft`/`fromRight` functions, but we'll be able to guarantee they are being used in a safe way.

**Comment by jdegoes** [comment]
:+1: (For something like this, anyway, details TBD.)

Yeah, I originally thought, "the user can unify the state themselves", but that would put unsafeness in userland code. Better to keep it in the installers, if we have to have it at all.

I think you'd want a `Tuple` of state and not an input.

**Comment by garyb** [comment]
Resolved an alternative way by #169
